### ISSUE TITLE
Needs update: CUT off country TAGS for correct lookup on TVMaze.tcl

### ISSUE THREAD

**Issue by terra-dude** [pzs-ng/pzs-ng#46]
Seems some releases that are tagged with US,UK,AU,CA,NZ in there directory name, get the wrong tvmaze lookup.

See example below:

org. relsname The.Zoo.US.S04E04.Beloved.Bugs.720p.HDTV.x264-GROUPNAME

**Wrong episode lookup**

!tv The.Zoo.US.S04E04.Beloved.Bugs.720p.HDTV.x264-GROUPNAME
TV-INFO Wake Up, Girl Zoo!(Comedy / Anime) Show Type: Animation - Rating: N/A 
TV-INFO Last Episode: S01E10: Episode 10 (2015-02-25) - Next Episode: N/A: N/A (N/A) - ShowStatus:Ended
TV-INFO TVMaze: https://www.tvmaze.com/shows/16159/wake-up-girl-zoo

**Correct episode lookup**

!tv The.Zoo.S04E04.Beloved.Bugs.720p.HDTV.x264-GROUPNAME
TV-INFO The Zoo: Beloved Bugs (Nature) Aired: 2020-04-18
TV-INFO Link: http://www.tvmaze.com/episodes/1835348/the-zoo-4x04-beloved-bugs

is there way to update the code on tvmaze.tcl for this.?

**Comment by terra-dude** [comment]
I have edit the following line in tvmaze.tcl

from: regsub -all -- {[\._]} $show_str " " show_str 

to: regsub -all -- {[\._]|US|UK|AU|CA|NZ} $show_str " " show_str 

do not know if it is correct?, otherwise please suggestions

**Comment by sirmarksalot** [comment]
Give these changes a try.

I've been running these changes for a few years.
The current script only gets the top result from the query.
My changes query the api and it gives all the top matches.
The script then goes threw each result and trys to match Year and Country code ISO based on the input dirname
https://github.com/sirmarksalot/pzs-ng/blob/tvmaze_lookup_fix/sitebot/plugins/TVMaze.tcl


```
!tv The.Zoo.US.S04E04.2014.Beloved.Bugs.720p.HDTV.x264-GROUPNAME
[tv-info] The Zoo: Beloved Bugs US/Documentary (Nature) Aired: 2020-04-18
[tv-info] URL: https://www.tvmaze.com/episodes/1835348/the-zoo-4x04-beloved-bugs Rating: 8.1
```

**Comment by teqnodude** [comment]
The fix by terra-dude seem to work just fine and is far simpler. 

Actually I've optimized the setting to avoid future problems with countries from this
`regsub -all -- {[\._]|US|UK|AU|CA|NZ|20[0-9][0-9]} $show_str " " show_str`

To
`regsub -all -- {[\._]|[A-Z][A-Z]|20[0-9][0-9]} $show_str " " show_str`

And get
```
!tv The.Zoo.US.S04E04.2014.Beloved.Bugs.720p.HDTV.x264-GROUPNAME
TV-INFO: The Zoo: Beloved Bugs (Nature) Aired: 2020-04-18
TV-INFO: Link: http://www.tvmaze.com/episodes/1835348/the-zoo-4x04-beloved-bugs

```

**Comment by sirmarksalot** [comment]
@teqnodude you are not wrong that terra's fix is much simpler but i posted mine as its a more complete fix to the issue at hand.
take this for example.

!tv is my changes
!ntv is pzs-ng master branch
!stv is ```regsub -all -- {[\._]|[A-Z][A-Z]|20[0-9][0-9]} $show_str " " show_str```

 ```
 !tv Shark.Tank.AU.S04E13.HDTV.x264-GRP
 [tv-info] Shark Tank: Episode 13 AU/Reality (N/A) Aired: 2018-08-07
 [tv-info] URL: https://www.tvmaze.com/episodes/1508279/shark-tank-4x13-episode-13 Rating: null
```

```
 !ntv Shark.Tank.AU.S04E13.HDTV.x264-GRP
 TVMaze Error :: No results found for "Shark Tank AU"
```

```
 !stv Shark.Tank.AU.S04E13.HDTV.x264-GRP
 [tv-info] Shark Tank: Episode 13 US/Reality (Drama) Aired: 2013-01-11
 [tv-info] URL: https://www.tvmaze.com/episodes/31928/shark-tank-4x13-episode-13 Rating: 7.4
```

**Comment by teqnodude** [comment]
@sirmarksalot You have a working tcl that you could link to here? The version you refer to in the comment above does not work at all.

**Comment by sirmarksalot** [comment]
@teqnodude try this https://github.com/sirmarksalot/pzs-ng/blob/tvmaze_lookup_fix/sitebot/plugins/TVMaze.tcl

The original comment months ago was missing a commit from pzs-ng master that fixed a function when using tcl8.6. That commit is now included in my fork

**Comment by teqnodude** [comment]
@sirmarksalot It spit out error `TVMaze Error :: missing value to go with key` when doing this: `!tv Origins.NZ.S01E02`

**Comment by teqnodude** [comment]
@sirmarksalot Same error when trying to make the dir `Origins.NZ.S01E02`. Making the dir `Origins.S01E02` works without a problem where the announce occur but not for NZ for some reason.

**Comment by sirmarksalot** [comment]
@teqnodude  I see the issue now and I have corrected it.
https://github.com/sirmarksalot/pzs-ng/blob/tvmaze_lookup_fix/sitebot/plugins/TVMaze.tcl
only issue now it I'm getting no result found. Is this show listed on TVMaze.com? I tried looking for it but was unable to find it.


``` 
!tv Origins.NZ.S01E02
TVMaze Error :: No results found for "Origins"
```


EDIT: I just pushed a change that adds support for dated release names

```
!tv Conan.2020.12.14.Jay.Pharoah.1080p.WEB.H264
[tv-info] Conan: Jay Pharoah US/Talk Show (Comedy) Aired: 2020-12-14
[tv-info] URL: https://www.tvmaze.com/episodes/1983505/conan-2020-12-14-jay-pharoah Rating: 7.5

!tv James.Corden.2020.12.14.Kyle.Chandler.WEB.h264
[tv-info] The Late Late Show with James Corden: Kyle Chandler, Meghan Trainor US/Talk Show (Comedy / Music) Aired: 2020-12-14
[tv-info] URL: https://www.tvmaze.com/episodes/1984447/the-late-late-show-with-james-corden-2020-12-14-kyle-chandler-meghan-trainor Rating: 5.5

!tv Seth.Meyers.2018.06.20.Mike.Myers.1080p.WEB.x264
[tv-info] Late Night with Seth Meyers: Mike Myers, Alison Brie, Kacey Musgraves US/Talk Show (Comedy) Aired: 2018-06-20
[tv-info] URL: https://www.tvmaze.com/episodes/1479561/late-night-with-seth-meyers-2018-06-20-mike-myers-alison-brie-kacey-musgraves Rating: 5.6

!tv Tooning.Out.The.News.2020.12.11.Week.In.Review.1080p.WEB.h264
[tv-info] Tooning Out the News:  US/Animation (Comedy) Aired: 2020-12-11
[tv-info] URL: https://www.tvmaze.com/episodes/1982595/tooning-out-the-news-1x112-inside-the-hill Rating: null

```

**Comment by teqnodude** [comment]
@sirmarksalot What was the problem? Can you highlight the changes?

**Comment by teqnodude** [comment]
No, it's not listed on TVMaze.com so it should show up as _"no result found"_. Nice work.

**Comment by teqnodude** [comment]
Only cosmetic left is when `Rating: null` it should go by what is set by `set ${np}::zeroconvert(%tvmaze_show_rating)               "N/A"`

**Comment by sirmarksalot** [comment]
Ok, I've fixed that now

**Comment by teqnodude** [comment]
@sirmarksalot `!tv The.Graham.Norton.Show.S28E11.720p.HDTV.x264`
`TVMaze Error :: can't read "show_next_episode_number": no such variable`

**Comment by sirmarksalot** [comment]
Fixed in my lastest commit.
This is actually a bug/typo that has existed in TVMaze.tcl for 5 years when it was converted over to use TVMaze
https://raw.githubusercontent.com/sirmarksalot/pzs-ng/tvmaze_lookup_fix/sitebot/plugins/TVMaze.tcl

```
!tv The.Graham.Norton.Show.S28E11.720p.HDTV.x264
[tv-info] The Graham Norton Show: George Clooney, Michael Sheen, David Tennant, Viola Davis, Daisy May Cooper, Vanessa Kirby, Michael Ball, Alfie Boe GB/Talk Show (Comedy) Aired: 2020-12-18
[tv-info] URL: https://www.tvmaze.com/episodes/1974811/the-graham-norton-show-28x11-george-clooney-michael-sheen-david-tennant-viola-davis-daisy-may-cooper-vanessa-kirby-michael-ball-alfie-boe Rating: 8.7
```
